src/main/java/net/mcreator/ui/modgui/GameRuleGUI.java

```diff
@@ -175,7 +175,7 @@ private void updateDefaultValueUI() {
 	@Override protected void afterGeneratableElementStored() {
 		super.afterGeneratableElementStored();
 		modElement.clearMetadata();
-		modElement.putMetadata("type", "Number".equals(gameruleType.getSelectedItem()) ? "number" : "boolean");
+		modElement.putMetadata("type", "Number".equals(gameruleType.getSelectedItem()) ? "number" : "logic");
```

Comment:
You will need to make updater to update metadata of existing mod elements from boolean to logic

Also maybe consider using eg. `VariableElementTypeLoader.LOGIC.getName()` here to re-use new united names